java/Ice/greeter/server/src/main/java/com/zeroc/demos/greeter/Server.java

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) ZeroC, Inc.
+
+package com.zeroc.demos.greeter;
+
+import com.zeroc.demos.greeter.Chatbot;
+import com.zeroc.Ice.Communicator;
+import com.zeroc.Ice.ObjectAdapter;
+
+class Server {
+    public static void main(String[] args) {
+        // Create an Ice communicator to initialize the Ice runtime. The communicator is disposed before the program exits.
+        try (Communicator communicator = com.zeroc.Ice.Util.initialize(args)) {
+            Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+                communicator.shutdown();
```

Comment:
This is not correct since the shutdown hook terminates the application when it completes.

Reply:
Updated to use destroy. We should fix this in a follow up PR to used the new proposed pattern.